### PR TITLE
Always export final solution

### LINKED ISSUE #106: Saving final solution
We discussed adding the final solution (and indicator) in the solution dictionary several times but I'm not sure if the agreement was reached in the end - I couldn't find an open issue about this. But now I took a look at the ODE PR and it looks like the solutions dictionary there has all the exported solutions, including the final one, but not the initial one. And it's not obvious to me how this was done.

So I was wondering what's the official status on this? :)

### PR DESCRIPTION
**Pull request by ddundo**
Closes #106 

With this PR the final solution of each subinterval always gets exported, instead of the first one.

Also tagging @stephankramer and @acse-ej321 for awareness :)

### CODE REVIEW

**Review by jwallwork23 — CHANGES_REQUESTED** [review]
Thanks for this @ddundo.

Please could you use `reversed` instead of `np.flip`? Good to use Python build-ins rather than external packages, where possible.

Could you also do the same thing in `adjoint.py`?

**Comment by ddundo** [comment]
This should be ready for a re-review now @jwallwork23 :) 

I also made the change I mention in the above comment. It looked like a bug to me

File: goalie/adjoint.py [hunk 1]
```diff
@@ -336,7 +336,9 @@ def wrapped_solver(subinterval, ic, **kwargs):
                 # Update forward and adjoint solution data based on block dependencies
                 # and outputs
                 sols = self.solutions[field]
-                for j, block in zip(range(num_exports - 1), solve_blocks[::stride]):
+                for j, block in zip(
+                    range(num_exports - 1), reversed(solve_blocks[::-stride])
+                ):
```
Comment (jwallwork23) [inline goalie/adjoint.py:341]: Similarly here.

**Review by jwallwork23 — APPROVED** [review]
Thanks very much for this @ddundo! Your bugfix and test tweaks make sense to me.

Just two very minor requests. If you fix those then feel free to merge.

File: goalie/mesh_seq.py [hunk 2]
```diff
@@ -570,7 +570,9 @@ def solve_forward(self, solver_kwargs: dict = {}) -> AttrDict:
 
                 # Update solution data based on block dependencies and outputs
                 sols = self.solutions[field]
-                for j, block in zip(range(num_exports - 1), solve_blocks[::stride]):
+                for j, block in zip(
+                    range(num_exports - 1), reversed(solve_blocks[::-stride])
+                ):
```
[the remark this answers is not in the document]
Comment (ddundo) [reply goalie/mesh_seq.py]: Nice! Done in eb187c0bb681061888e0b9409e2c8ac56856d633

**Comment by ddundo** [comment]
Thanks Joe! I did as you suggested and will merge :)

**Pull request merged**